### PR TITLE
#6 - Indexer fails when syncing with btctest network

### PR DESCRIPTION
**Pull request by aph5nt**
this fixed the problem with too big ticks to datetime conversion.
the 'peek' has been replaced with 'last', because 'peek' will always get the first item from the queue and we were always getting the same values

### CODE REVIEW

File: NBitcoin.Indexer/IndexerTrace.cs [hunk 1]
```diff
@@ -162,43 +162,25 @@ public static void Processed(int height, int totalHeight, Queue<DateTime> lastLo
...
-                    var downloadedSize = GetSize(lastHeight, height);
-                    var remainingSize = GetSize(height, totalHeight);
-                    var estimatedTime = downloadedSize == 0.0m ? TimeSpan.FromDays(999.0)
-                        : TimeSpan.FromTicks((long)((remainingSize / downloadedSize) * time.Ticks));
-                    _Trace.TraceInformation("Blocks {0}/{1} (estimate : {2})", height, totalHeight, Pretty(estimatedTime));
+                    var lastHeight = lastHeights.Last();
+                    var time = DateTimeOffset.UtcNow - lastLogs.Last();
+                    var remaining = totalHeight - height;
+                    var block = height - lastHeight;
+                    var blocksRemaining = remaining/block;
```
Comment (NicolasDorier) [inline NBitcoin.Indexer/IndexerTrace.cs]: line would crash if height == lastHeight.

**Comment by NicolasDorier** [comment]
thanks a lot, I am surprised, normally peek take the next element that will be unqueued. 

I'll merge that very soon, sadly I need to get my PC back on its feet to release the nuget package, should be done in 1 or 2 days.

**Comment by NicolasDorier** [comment]
hey sorry for taking so much time.
For the time approximation you can&t use just the height because first blocks are order of magnitude faster than later one.

**Comment by NicolasDorier** [comment]
I fixed the bug, however I am surprised that the peek is always 0. I am queuing heights and peek should returns the next height to be removed.

**Comment by aph5nt** [comment]
@NicolasDorier ok, thats good that a bug has been fixed :)

**Pull request closed without merging**